### DISCUSSION TITLE
https://saicharankandukuri.github.io/posts/termux-signal9-fix/

### DISCUSSION THREAD

**Discussion by giscus** [SaicharanKandukuri/SaicharanKandukuri.github.io#3] [Blog]
# https://saicharankandukuri.github.io/posts/termux-signal9-fix/

fix termux signal 9 error without external PC

https://saicharankandukuri.github.io/posts/termux-signal9-fix/

**Comment by AnninoDr4** [comment]
Finally no need of PC
> it works

**Reply by ayon-ssp** [reply]
![img](https://c.tenor.com/xp7YJ6De7P4AAAAM/explain-it-anthony-mennella.gif)
In Class...🥲..Cause No time to read😆🤣.

**Reply by SaicharanKandukuri** [reply]
No time need to write more!

**Comment by Cosmos-Kramer** [comment]
Awesome!  This is excellent, thanks very much!

In case anybody's experience matches mine, FYI:

1. After pairing (adb pair ....etc.), it wasn't necessary to run the command    adb connect localhost:<debug port>    amd I don't believe it should be (?).

2. When running the final adb command, if it fails with...
> adb: more than one device/emulator
...you just need to specify which device adb is committing the change to, using   *-s devicename*   where *devicename* will be like localhost:12345.  Just get your device name by running   *adb devices*  then update the original adb command to include it, ike so:
  adb -s localhost:12345 \
    shell \
    etc. etc.

(Thank you again!)

**Reply by Cosmos-Kramer** [reply]
**Preview is your friend!  Sorry, let's try that again:**

In case anybody's experience matches mine, FYI:

1. After pairing (adb pair ....etc.), it wasn't necessary to run the command    adb connect localhost:<debug port>    amd I don't believe it should be (?).

2. When running the final adb command, if it fails with...
> adb: more than one device/emulator

...you just need to specify which device adb is committing the change to, using   *-s devicename*   where *devicename* will be like localhost:12345.  Just get your device name by running   *adb devices*  then update the original adb command to include it, ike so:

adb -s localhost:12345 \\
 shell \\
 etc. etc.

(Thank you again!)

**Comment by Mopwater94** [comment]
Dose this process work for android 13, 64 bit

**Reply by Cosmos-Kramer** [reply]
Absolutely.  The change (the problem) being addressed was first introduced in Android 12, and there's every reason to expect it will remain present in Android for years (and versions) to come -- but at the very least,.it applies to Android 12 / 13 / 14 so far.

**Comment by SaicharanKandukuri** [comment]
**Update:**

here is a video i made to show a rather easy to fix this issue: https://www.youtube.com/watch?v=GCN0gh1yXSs

**Comment by MRGABOZ** [comment]
Thank you. One question in case you want to leave it as it is before, what command should I use?

**Reply by SaicharanKandukuri** [reply]
if you are following youtube video then use this command after executing `rish`
```bash
device_config set_sync_disabled_for_tests none
```

**for pc**
```bash
adb shell device_config set_sync_disabled_for_tests none
```

**Reply by SaicharanKandukuri** [reply]
just reboot after that and it goes back to normal

**Reply by MRGABOZ** [reply]
Ohh ok thank you very much for the tutorial hehe

**Comment by Banned420** [comment]
It says pair was successful but when I run adb devices the device list is empty and I can't run any commands. Any help would be appreciated. Thanks

**Comment by EliRoberts5** [comment]
awesome, seemed to work perfectly ( android 15 phantom process fix )
you can test the result with 
adb \
    shell \
    device_config \
    get activity_manager \
    max_phantom_processes

I didn’t need the github repo I just installed platform tools and completed adb cpnfig and commands
if this continues to work it loks like an awesome fix to keep termux etc running well , which is really important, thanks again 👍

**Reply by SaicharanKandukuri** [reply]
Btw, android 14+ has new option to fix **signal 9** issue.

This video covers it good https://youtu.be/N5Q5J36wIkc?t=93

**Reply by Cosmos-Kramer** [reply]
Ahhh this is perfect!!    Thanks so much for the follow-up!!  \:)
